src/services/ServiceMappingService.ts

```diff
@@ -106,31 +112,61 @@ export class ServiceMappingService {
     }
 
     private getQueryParameters(parameters: IOpenAPI3Parameter[] | undefined, models: IModelsContainer): IMethodParameterModel[] {
-        return (
-            parameters?.map((z) => {
-                const parameter = {
-                    name: lowerFirst(z.name),
-                    optional: z.required === undefined ? false : !z.required,
-                    place: ParameterPlace.Query,
-                    isCollection: false,
-                    dtoType: '',
-                    isModel: false
-                };
-
-                if (this.typesGuard.isSimple(z.schema)) {
-                    parameter.dtoType = this.typesService.getSimpleType(z.schema).dtoType;
-                    return parameter;
-                }
-
-                if (this.typesGuard.isReference(z.schema)) {
-                    parameter.dtoType = this.findModel(models, z.schema)?.dtoType ?? '';
-                    parameter.isModel = true;
-                    return parameter;
-                }
+        if (parameters === undefined) {
+            return [];
+        }
+
+        return parameters.filter(z => z.in === 'query').map((z) => {
+            const parameter = {
+                name: lowerFirst(z.name),
+                optional: z.required === undefined ? false : !z.required,
+                place: ParameterPlace.Query,
+                isCollection: false,
+                dtoType: '',
+                isModel: false
+            };
+
+            if (this.typesGuard.isSimple(z.schema)) {
+                parameter.dtoType = this.typesService.getSimpleType(z.schema).dtoType;
+                return parameter;
+            }
+
+            if (this.typesGuard.isReference(z.schema)) {
+                parameter.dtoType = this.findModel(models, z.schema)?.dtoType ?? '';
+                parameter.isModel = true;
+                return parameter;
+            }
+
+            return parameter;
+        });
+    }
+
+    private getPathParameters(parameters: IOpenAPI3Parameter[] | undefined, methodName: string): IMethodParameterModel[] {
+        if (parameters === undefined) {
+            return [];
+        }
+
+        return parameters.filter(z => z.in === 'path').map((z) => {
+            const parameter = {
+                name: lowerFirst(z.name),
+                optional: false,
+                place: ParameterPlace.Path,
+                isCollection: false,
+                dtoType: '',
+                isModel: false
+            };
 
+            if (this.typesGuard.isSimple(z.schema)) {
+                parameter.dtoType = this.typesService.getSimpleType(z.schema).dtoType;
                 return parameter;
-            }) ?? []
-        );
+            }
+
+            if (this.typesGuard.isReference(z.schema)) {
+                throw new Error(`Method ${methodName} contains reference type in path parameters, which is not allow`);
+            }
```

Comment:
most of it is copy paste from previous method

maybe we should create separate objects for parameters e.g. MethodParameter and (PathMethodParameter, QueryMethodParameter) 
and initilize from IOpenAPI3Parameter

Reply:
Done